### ISSUE TITLE
Pin Apps (even when they dont "support" links)

### ISSUE THREAD

**Issue by yant0** [TrianguloY/UrlChecker#296]
### Describe a related problem (optional)

An option to pin an app that is present on the dropdown menu. Useful for downloaders which may or may not have websites configured to open it. Or other use case where user frequent an app but it doesnt show it on the dropdown and had to press the share button to open legacy share menu

### Describe your suggested feature

A dropdown, positioned at the very bottom

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by TrianguloY** [comment]
If an app is not shown on the dropdown then it probably won't support 'opening' the url, and the share is the only way to use it.

Another possible issue is that the specific link is 'owned' by an app (like youtube links owned by the youtube app) and those not only have preference, but disable every other app from opening it (by default). Luckily you can change this, you should open the android settings for the app that is opening the links, search for the options 'Open Supported Links' and set as 'ask every time' (or similar). At the same time, in the app that is not opening the links (and you wanted), set it too.
If the app is still not opening the links, most probably it wasn't designed to open them in the first place, and you should use the share option.

Android apps can request data with different intents, and the open and share intents are different so one app that can open may not know what to do when it is shared, and vice versa.

If your issue is with a specific app and link, post them here and we can try to investigate that particular scenario.

**Comment by yant0** [comment]
> Another possible issue is that the specific link is 'owned' by an app (like youtube links owned by the youtube app) and those not only have preference, but disable every other app from opening it (by default).

Yeah i was thinking with this in mind. Most of the time when you open links to youtube, you probably want to watch it. I personally put left youtube links to youtube, but you can share the link within the app and that opens UrlChecker. But with that, some apps just dont add which links they support. For me, i use [Seal](https://github.com/JunkFood02/Seal) which doesnt support any links and doesnt show in UrlChecker.

> Android apps can request data with different intents, and the open and share intents are different so one app that can open may not know what to do when it is shared, and vice versa.

I was wondering about this too. But i think just passing it as text would suffice (since Seal strips anything beside the link) but i dont know if other people would say the same.

Also, i dont know how UrlChecker pick the apps on the dropdown, but when using the legacy share menu it shows all the apps that would know what to do with the link. im sure this would benefit people who know what they're doing

**Comment by TrianguloY** [comment]
> Also, i dont know how UrlChecker pick the apps on the dropdown, but when using the legacy share menu it shows all the apps that would know what to do with the link. im sure this would benefit people who know what they're doing

That's what I was trying to explain. The dropdown is populated with the apps that can **open** the link. The share menu is different and contains a totally different set of apps. You cannot mix both.

Android allows you to open a native dialog to handle an intent. Right now the share option uses this, but the open option does not, and instead populates the dropdown.

There has been talks about splitting the open and share functionalities into different modules, which could allow the share module to also use a dropdown instead of the native popup. The pinning apps option has also been mentioned before.

**Comment by PabloOQ** [comment]
#47 might be relevant. In this [comment](https://github.com/TrianguloY/UrlChecker/issues/47#issuecomment-1409147608) at the very bottom, if you click on share it shows a mockup I made on how share and open with could co exist together. Using both "open with" and "share", in the same menu would easily solve this problem, by sneaking "share to Seal" in between the other apps.
